maploc/data/dataset.py

```diff
@@ -126,44 +127,74 @@ def get_view(self, idx, scene, seq, name, seed, bbox_tile):
         cam_dict = self.data["cameras"][scene][seq][self.data["camera_id"][idx]]
         cam = Camera.from_dict(cam_dict).float()
 
-        if "roll_pitch_yaw" in self.data:
-            roll, pitch, yaw = self.data["roll_pitch_yaw"][idx].numpy()
-        else:
-            roll, pitch, yaw = decompose_rotmat(self.data["R_c2w"][idx].numpy())
         image = read_image(self.image_dirs[scene] / (name + self.image_ext))
 
-        if "plane_params" in self.data:
-            # transform the plane parameters from world to camera frames
-            plane_w = self.data["plane_params"][idx]
-            data["ground_plane"] = torch.cat(
-                [rotmat2d(deg2rad(torch.tensor(yaw))) @ plane_w[:2], plane_w[2:]]
-            )
         if self.cfg.force_camera_height is not None:
             data["camera_height"] = torch.tensor(self.cfg.force_camera_height)
         elif "camera_height" in self.data:
             data["camera_height"] = self.data["height"][idx].clone()
 
         # raster extraction
         canvas = self.tile_managers[scene].query(bbox_tile)
-        xy_w_gt = self.data["t_c2w"][idx][:2].numpy()
-        uv_gt = canvas.to_uv(xy_w_gt)
-        uv_init = canvas.to_uv(bbox_tile.center)
         raster = canvas.raster  # C, H, W
 
+        world_T_cam = Transform3D.from_Rt(
+            self.data["R_c2w"][idx], self.data["t_c2w"][idx]
+        ).float()
+        world_T_cam2d = Transform2D.from_Transform3D(
+            world_T_cam
+        )  # angle is b/w world x and cam z
+
+        # gcam: gravity-aligned camera with z=optical axis
+        # gcamxyz: gcam rotated such that z:up,x:right,y:forward.
+
+        gcam_angle = world_T_cam2d.angle - 90
+        Rz = Transform2D.from_degrees(gcam_angle, torch.zeros(2))
+        world_R_gcamxyz = torch.eye(3)
+        world_R_gcamxyz[:2, :2] = Rz.R
+        world_T_gcamxyz = Transform3D.from_Rt(world_R_gcamxyz, world_T_cam.t).float()
+        gcamxyz_T_gcam = Transform3D.from_Rt(
+            Rotation.from_euler("X", -90, degrees=True).as_matrix(), torch.zeros(3)
+        ).float()
+        world_T_gcam = world_T_gcamxyz @ gcamxyz_T_gcam
+        gcam_T_cam = world_T_gcam.inv() @ world_T_cam
+
+        world_T_tile = Transform2D.from_Rt(torch.eye(2), canvas.bbox.min_).float()
+        tile_T_cam = world_T_tile.inv() @ world_T_cam2d
+        map_T_cam = Transform2D.from_Rt(
+            tile_T_cam.R, tile_T_cam.t * canvas.ppm - 0.5
```

Comment:
Maybe add a `Transform2D.to_pixels` function since this operation is done a bunch of times?